src/utils/saas.js

```diff
@@ -0,0 +1,199 @@
+ * @param {string} name - Identifier for the request.
+ * @param {string} url - Target URL.
+ * @param {object} req - Request options and headers.
+ * @param {number} timeout - Timeout in milliseconds.
+ * @returns {Promise<any|string|null>} - Response data or null.
+ */
+export async function request(name, url, req = [], timeout = 60000) {
+  if (timeout <= 0 || timeout > 300000) { // max 5 minutes
+    throw new Error('Timeout must be between 1ms and 300000ms');
+  }
+
+  // allow requests for 60s max
+  const abortController = new AbortController();
+  const abortTimeout = setTimeout(() => abortController.abort(), timeout);
+
+  const headers = {
+    ...req.headers,
+    'User-Agent': 'Spacecat/1.0',
+  };
+
+  const resp = await fetch(url, {
+    ...req,
+    headers,
+    signal: abortController.signal,
+  });
+  // clear the abort timeout if the request passed
+  clearTimeout(abortTimeout);
+
+  let responseText = '';
+
+  if (resp.ok) {
+    if (resp.status < 204) {
+      // ok with content
+      return resp.headers.get('content-type') === 'application/json' ? resp.json() : resp.text();
+    } else if (resp.status === 204) {
+      // ok but no content
+      return null;
+    }
+  } else {
+    try {
+      responseText = await resp.text();
+      // eslint-disable-next-line no-unused-vars
+    } catch (e) { /* nothing to be done */ }
+  }
+
+  throw new Error(`Request '${name}' to '${url}' failed (${resp.status}): ${resp.headers.get('x-error') || resp.statusText}${responseText.length > 0 ? ` responseText: ${responseText}` : ''}`);
+}
+
+/**
+ * Fetches spreadsheet data from a given config path and optional sheet name.
+ * Useful for loading configuration or data from remote sources.
+ * @param configPath - The URL or path to the spreadsheet config.
+ * @param sheet - Optional sheet name to request specific data.
+ * @returns {Promise<*|string|null>} - The spreadsheet data, string, or null.
+ */
+export async function requestSpreadsheet(configPath, sheet) {
+  return request(
+    'spreadsheet',
+    configPath + (sheet ? `?sheet=${sheet}` : ''),
+  );
+}
+
+/**
+ * Validates the required fields in the config object for a given locale.
+ * @param config - Configuration object to validate.
+ * @param locale - Locale identifier.
+ * @returns {*} - The validated config object.
+ */
+export function validateConfig(config, locale) {
+  const requiredConfigFields = [
+    'commerce-customer-group',
+    'commerce-environment-id',
+    'commerce-store-code',
+    'commerce-store-view-code',
+    'commerce-website-code',
+    'commerce-x-api-key',
+  ];
+  const missingFields = [];
+
+  for (const field of requiredConfigFields) {
+    if (!config[field]) {
+      missingFields.push(`Missing required parameter: ${field}`);
+    }
+  }
+
+  if (missingFields.length > 0) {
+    throw new Error(`Missing required config parameters for ${locale} locale: ${missingFields.join(', ')}`);
```

Comment:
Please check the docs. There is more than one configuration format, depending on the deployment model. https://experienceleague.adobe.com/developer/commerce/storefront/setup/configuration/commerce-configuration/